### ISSUE TITLE
Load missing map parts from online map

### ISSUE THREAD

**Issue by user123456000** [thebuildcraft/MapLink#5]
Although I understand that this is a very difficult task that is beyond the scope of this mod, but what about loading the missing map parts from an online map?

**Comment by thebuildcraft** [comment]
The idea is good... But as you already said: It would be very difficult. Xaero's World Map doesn't save the data as images. But the online maps just give images. I would need to convert every pixel into a block type. To do this, the code would need to compare an online-map-part with a local part of the xaero map to get some sort of dictionary to convert the pixel color into blocks.

If you know of any good way how to do it, I will consider it. (But only after I got the multiloader setup with architectury working...)
If you don't have a good idea how to do it, I will close the issue for now and will maybe look into it some day... no promise though

**Comment by user123456000** [comment]
It would be possible to place an image from an online map under the mod map. Yes, the gap between the parts from the online map and the mod map would be striking, but even this would be a great achievement

**Comment by thebuildcraft** [comment]
That could work. But I dont know how hard this would be, because I would probably need to reverse-engineer the rendering code for xaeros map to place the images in the right place. (and xaeros map is not open source)
Maybe some day I will look into it.

**Comment by sametcetinkaya1447** [comment]
hello i dont know how to make an idea so i am typing from here could u add the  delete waypoints if i disable the mod i am being have to delete all manually

**Comment by thebuildcraft** [comment]
Thanks for letting me know... it must have broken it without realizing because it definitely worked at some point. You can create your own issue by clicking the green "New Issue" button here: https://github.com/thebuildcraft/RemotePlayerWaypointsForXaero/issues
I will fix it. But I don't know when the fix will be released. For now, you can keep the mod enabled and disable player and marker waypoints instead.

**Comment by SPLATPLAYS** [comment]
Any more updates on this?

**Comment by FoZy-STYLE** [comment]
> If you know of any good way how to do it, I will consider it. (But only after I got the multiloader setup with architectury working...) If you don't have a good idea how to do it, I will close the issue for now and will maybe look into it some day... no promise though


You can compromise on the detail for faster checking.
I don't think that you should make it check and convert every single block. 
I would make a color range map for each type of block and have the mod detect it from the grid in the image map. For example, if the pixel chosen inside of the voxel matches the range of stone (grayish), then the mod will register that as simply stone even if it was actually andesite or stone bricks or even gray wool since unless you zoom in its very hard to notice the difference and they all look virtually the same.
You could have for example quartz block for all white blocks in the map, stone for gray, red for mangrove or terracotta, orange for red sand, pale yellow for sand etc.

**Comment by jbm11208** [comment]
This feature would not have to run all the time, there could be an option to port over the online map using one of the previously mentioned ideas to fill in all the empty spots on Xaero's, and the data will not update unless the option to port over the online map data is clicked again. as the player loads the empty areas on their client, the map will be filled with genuine data from Xaero's, and those spots will not be pulled from the online map.

**Comment by thebuildcraft** [comment]
Surprise: I did it ... well it's Bluemap only for this update, but will be expanded in future updates :)

**Comment by user123456000** [comment]
Wow, that's awesome. The only thing is, it would be nice if it had some kind of progress bar so I could see how much has already been converted and how much is left.

**Comment by thebuildcraft** [comment]
I wanted to keep it open because there is still Dynmap, Pl3xMap, SquareMap, a UI and progress bar missing.
(I will get around to do them eventually... 😀)
